src/main/scala/domain/CustomerService.scala

```diff
@@ -0,0 +1,10 @@
+package domain
+
+import cats.MonadThrow
+import domain.CustomerRepository.CustomerRepository
+
+class CustomerService[F[_]](repository: CustomerRepository[F]) {
```

Comment:
when u migrate CustomerInMemoryRepository to correct type it will require to migrate CustomerRepository to correct type and it will require CustomerService

generally: all should have MonadThrow as a bound

u can think on this like that: do u think that domain service is usable if it cannot produce any errors on its own?